django/contrib/admin/options.py

```diff
@@ -473,9 +473,12 @@ def to_field_allowed(self, request, to_field):
             for inline in admin.inlines:
                 registered_models.add(inline.model)
 
-        for related_object in (opts.get_all_related_objects(include_hidden=True) +
-                               opts.get_all_related_many_to_many_objects()):
-            related_model = related_object.model
+        related_objects = (
+            f for f in opts.get_fields(include_hidden=True)
+            if (f.auto_created and not f.concrete)
```

Comment:
like above, would it work to do: `f.one_to_one and f.one_to_many and not f.concrete`?